modules/Bio/EnsEMBL/Hive/RunnableDB/FastaFactory.pm

```diff
@@ -167,15 +177,22 @@ sub write_output {
             $chunk_length   = 0;
             $chunk_size     = 0;
             $chunk_number++;
-            $chunk_name     = $output_prefix.$chunk_number.$output_suffix;
+            
+            my $current_output_directory;
+            
             if ($self->param('hash_directories')) {
                 my $dir_tree = dir_revhash($chunk_number);
-                if ($dir_tree ne '') {
-                    mkpath($dir_tree);
-                    $chunk_name = $dir_tree.'/'.$chunk_name;
+                if ($dir_tree eq '') {                
+		  $current_output_directory = $output_dir;
+                } else {
+                  $current_output_directory = $output_dir . '/' . $dir_tree;
                 }
+                mkpath($current_output_directory);
+            } else {            
+	      $current_output_directory = $output_dir;
             }
-            $chunk_seqio    = Bio::SeqIO->new(-file => '>'.$chunk_name, -format => 'fasta');
+            $chunk_name   = $current_output_directory . '/' . $output_prefix . $chunk_number . $output_suffix;
+            $chunk_seqio = Bio::SeqIO->new(-file => '>' . $chunk_name, -format => 'fasta');
```

Comment:
This line seems identical in function to what was line 178. Can you restore the original indentation ?